apps/wallet-mobile/src/features/WalletManager/SelectWalletFromList/SelectWalletFromListScreen.tsx

```diff
@@ -146,20 +145,16 @@ const SupportTicketLink = () => {
 }
 
 const AddWalletButton = () => {
-  const navigation = useNavigation()
   const strings = useStrings()
   const {styles} = useStyles()
-  const {walletImplementationIdChanged, reset: resetSetupWallet} = useSetupWallet()
+  const {reset: resetSetupWallet} = useSetupWallet()
+  const {resetToWalletSetup} = useWalletNavigation()
 
   return (
     <Button
       onPress={() => {
         resetSetupWallet()
```

Comment:
I'm guessing resetSetupWallet -> re-initialize state and resetToWalletSetup navigates to it? may worth adding a comment on top